lib/msprime.c

```diff
@@ -297,11 +297,18 @@ msp_set_gene_conversion_tract_length(msp_t *self, double tract_length)
 {
     int ret = 0;
 
-    if (tract_length < 0 || tract_length > self->sequence_length) {
+    if (tract_length < 1 || tract_length > self->sequence_length) {
         ret = MSP_ERR_BAD_PARAM_VALUE;
         goto out;
     }
     self->gc_tract_length = tract_length;
+    if (tract_length == 1) {
+        self->continuous_gc_tract_length = 0.001;
```

Comment:
Is it really worth storing this? I doubt this computation will even show up on any profiles, so my vote would be to keep it simple and to compute this value on the fly in ``msp_generate_tract_length``

Reply:
I thought since potentially we might be generating lots of (noneffective) gc events, I do this computation only once.
However, I have no idea how much this influences the performance, so I agree and will compute it on the fly.

Reply:
Saving computations like this will only make a difference within really tight inner loops. Our runtime is dominated by the cost of maintaining the Fenwick trees, binary search tree lookups and the cost of iterating over segments in merge_ancestors. Something like this won't even show up.

Reply:
I see. So adding the variable gc rates, where we need another Fenwick tree will be much more of an issue regarding performance. Will compute it on the fly.